library_generation/utilities.sh

```diff
@@ -273,3 +273,19 @@ get_version_from_versions_txt() {
   version=$(grep "$key:" "${versions}" | cut -d: -f3) # 3rd field is snapshot
   echo "${version}"
 }
+
+detect_OS() {
```

Comment:
Judging by what the function does, the name should be `detect_os_architecture`

Reply:
makes more sense. Changed